### PR TITLE
adds circle config

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/2737. Note this is based on #1.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    - run: |
+        wget -q $(curl -sS -H "Authorization: token $RELEASE_TOKEN" https://api.github.com/repos/giantswarm/architect/releases/latest | grep browser_download_url | head -n 1 | cut -d '"' -f 4)
+        chmod +x ./architect
+        ./architect version
+    - run: ./architect build
+    - deploy:
+        command: |
+          if [ "${CIRCLE_BRANCH}" == "master" ]; then
+            ./architect deploy
+          fi
```
Comment (xh3b4sd) [inline .circleci/config.yml:17]: Adding the circle config. Standard.

File: main.go [hunk 2]
```diff
@@ -11,13 +12,31 @@ import (
...
 func main() {
+	if (len(os.Args) > 1) && (os.Args[1] == "version") {
+		d, err := yaml.Marshal(newVersionResponse())
+		if err != nil {
+			panic(err)
+		}
+		fmt.Printf("%s", d)
+
+		return
+	}
```
Comment (xh3b4sd) [inline main.go:38]: This is for architect and to be consistent.

File: main.go [hunk 3]
```diff
@@ -26,8 +45,10 @@ func main() {
 	}
 
 	mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		io.WriteString(w, "Hello World")
+		w.Header().Set("Content-Type", "application/json")
+		json.NewEncoder(w).Encode(newVersionResponse())
```
Comment (xh3b4sd) [inline main.go:49]: Responding with a version response like in other microservices.

File: version_response.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+func newVersionResponse() VersionResponse {
+	return VersionResponse{
+		Description: description,
+		GitCommit:   gitCommit,
+		GoVersion:   runtime.Version(),
+		Name:        name,
+		OSArch:      runtime.GOOS + "/" + runtime.GOARCH,
+		Source:      source,
+	}
+}
```
Comment (xh3b4sd) [inline version_response.go:23]: Adding the version response magic.

**Review by fgimenez — APPROVED** [review]
LGTM

**Pull request merged**